nima/webserver/webserver/src/main/java/io/helidon/nima/webserver/spi/ServerConnectionProvider.java

```diff
@@ -58,6 +59,21 @@ public interface ServerConnectionProvider {
      */
     ServerConnection connection(ConnectionContext ctx);
 
+    /**
+     * Provider's specific configuration nodes names.
+     *
+     * @return names of the nodes to request
+     */
+    Iterable<String> configKeys();
+
+    /**
+     * Provider's configuration reader.
+     * Node with {@code configKey()} name will be provided.
+     *
+     * @param config {@link Config} configuration node
+     */
+    void config(Config config);
```

Comment:
I think the config key should be a second parameter here, as it may not be possible to obtain it from the config instance (e.g. if you `detach()` the config)

Reply:
This method is being called from WebServer. Builder connectionProviders() method at the beginning of server lifecycle so I was not expecting any problems with the key.
Also processing of nodes uses names only just to walk trough provided node subtree, not fully-qualified keys.
But I added the key here.